### DISCUSSION TITLE
[Feature]: Comments

### DISCUSSION THREAD

**Discussion by PixelJonas** [immich-app/immich#1660] [Feature Request]
### Feature detail

Add a feature, so people can Like and Comment on photos in a shared album.

### Platform

Mobile App

**Comment by palitu** [comment]
as an addition to this, i see this working in this manner:

1. A comment, like or tag is added to a photo, via the WebUI or App
2. The details are embedded in the photo's meta-data
2.1 The details written should be easily identifiable that they were written by immich (ie, tag is `immich/<tag details>`, comment is `immich-<comment string>`)
2.2 when displaying these details the `immich-` prefix is hidden.
4. The details are recorded/sync'd into the DB (for speed of searching)

These are then utilised in the following way:

**Comments**
- Are visible under the 
   - thumbnail
   - image 
   - image information panel
- Comments are searchable

**Liked**
 - shown in the image thumbnails
 - shown in the title bar of the image
 - can be filtered for all likes (a logical album)
 - can be used as a search criteria (ie `liked` AND tree)

**Tags**
 - shown in title bar  of image
 - shown in info bar of image
 - can be filtered for an individual/set of tags (ie, logical albums)
 - can be used as a search criteria

To be honest, this is one of the massive things that myself and my friends are chasing to round out our requirements for a photo viewer/backup/manager, and it will make it infinitely more useful as a catch all photo manager for us.

**Comment by palitu** [comment]
note, as far as the ML type automated tagging, it would be good if these are also embedded in the image, but also under a prefix.

The reason for the prefix is to track where the data comes from, and to be able to cleanse it out automatically if desired.

**Comment by digiprospector** [comment]
so, does this feature supported now?

**Reply by alextran1502** [reply]
Not yet

**Reply by dekiesel** [reply]
Without asking for an eta: is it planned/on the roadmap somewhere?

**Comment by irrigate-hanks** [comment]
+1 for this. I have a longstanding album of my children with comments from grandparents and great-grandparents watching them grow. Obviously a 1-1 transfer option probably isn't realistic, but just having comments at all would allow me to manually replicate these so these messages being left aren't solely reliant on Google's platform.

**Comment by martabal** [comment]
That would be a great future ! I will work on it after the FR improvements. Are reactions better than likes (not only `❤️` but also all emojis) ?

**Reply by PixelJonas** [reply]
while I like the idea, I personally don't share my albums with that many people that different reactions would make a big difference. Depending on the complexity of implementation I'm totally fine with a single-emoji reaction. If we are going for multiple emojis, then I'd prefer to have a single-tap reaction of a default (changeable?) emoji and be able to switch the emoji (long-press?). My 2 cents

**Comment by mmorys** [comment]
Would love to see this implemented. I am looking to replace a Google shared album where friends and family can like or comment on photos.

I also like Google Photos timeline view where you can see the history of what photos people commented on and liked.

**Reply by mmorys** [reply]
Implementing in such a way that in the future notifications could also be triggered with comments/reactions would be awesome.

**Comment by fivestones** [comment]
This is implemented in v1.84.0.

**Comment by Sums123** [comment]
Adding to this
https://github.com/immich-app/immich/discussions/1660#discussioncomment-6708483

I think tagging (as in using @ in the comment) other users would be nice

**Comment by SLFaleiro** [comment]
Any update on this?

**Comment by petrasl1976** [comment]
I would love my family member could comment on shared photos. It would be even more cooler to read those comments after a while!

**Reply by Soluce-Technologies** [reply]
Any update about this feature ?

**Comment by bo0tzz** [comment]
This has been supported for forever.
